contracts/src/v0.8/dev/vrf/testhelpers/VRFV2PlusWrapperLoadTestConsumer.sol

```diff
@@ -0,0 +1,175 @@
+      emit WrapperRequestMade(requestId, paid);
+    }
+  }
+
+  function fulfillRandomWords(uint256 _requestId, uint256[] memory _randomWords) internal override {
+    require(s_requests[_requestId].paid > 0, "request not found");
+    uint256 fulfilmentBlockNumber = ChainSpecificUtil.getBlockNumber();
+    uint256 requestDelay = fulfilmentBlockNumber - requestHeights[_requestId];
+    uint256 requestDelayInMillions = requestDelay * 1_000_000;
+
+    if (requestDelay > s_slowestFulfillment) {
+      s_slowestFulfillment = requestDelay;
+    }
+    s_fastestFulfillment = requestDelay < s_fastestFulfillment ? requestDelay : s_fastestFulfillment;
+    s_averageFulfillmentInMillions = s_responseCount > 0
+      ? (s_averageFulfillmentInMillions * s_responseCount + requestDelayInMillions) / (s_responseCount + 1)
+      : requestDelayInMillions;
+
+    s_responseCount++;
+    s_requests[_requestId].fulfilled = true;
+    s_requests[_requestId].randomWords = _randomWords;
+    s_requests[_requestId].fulfilmentTimestamp = block.timestamp;
+    s_requests[_requestId].fulfilmentBlockNumber = fulfilmentBlockNumber;
+
+    emit WrappedRequestFulfilled(_requestId, _randomWords, s_requests[_requestId].paid);
+  }
+
+  function getRequestStatus(
+    uint256 _requestId
+  )
+    external
+    view
+    returns (
+      uint256 paid,
+      bool fulfilled,
+      uint256[] memory randomWords,
+      uint requestTimestamp,
+      uint fulfilmentTimestamp,
+      uint256 requestBlockNumber,
+      uint256 fulfilmentBlockNumber
+    )
+  {
+    require(s_requests[_requestId].paid > 0, "request not found");
+    RequestStatus memory request = s_requests[_requestId];
+    return (
+      request.paid,
+      request.fulfilled,
+      request.randomWords,
+      request.requestTimestamp,
+      request.fulfilmentTimestamp,
+      request.requestBlockNumber,
+      request.fulfilmentBlockNumber
+    );
+  }
+
+  function reset() external {
+    s_averageFulfillmentInMillions = 0; // in millions for better precision
+    s_slowestFulfillment = 0;
+    s_fastestFulfillment = 999;
+    s_requestCount = 0;
+    s_responseCount = 0;
+  }
+
+  /// @notice withdrawLink withdraws the amount specified in amount to the owner
+  /// @param amount the amount to withdraw, in juels
+  function withdrawLink(uint256 amount) external onlyOwner {
+    LINK.transfer(owner(), amount);
+  }
+
+  /// @notice withdrawNative withdraws the amount specified in amount to the owner
+  /// @param amount the amount to withdraw, in wei
+  function withdrawNative(uint256 amount) external onlyOwner {
+    (bool success, ) = payable(owner()).call{value: amount}("");
+    require(success, "withdrawNative failed");
+  }
+
+  function getWrapper() external view returns (VRFV2PlusWrapperInterface) {
+    return VRF_V2_PLUS_WRAPPER;
+  }
+
+  receive() external payable {}
+
+  function getBalance() public view returns (uint) {
+    return address(this).balance;
+  }
```

Comment:
Just a note for future. these methods seem to belong in the base wrapper consumer. I think we can revisit after the on-going discussions around wrapped native support.

Reply:
Fixed in this PR: https://github.com/smartcontractkit/chainlink/pull/10966